internal/consensus/replay.go

```diff
@@ -240,10 +240,17 @@ func (h *Handshaker) NBlocks() int {
 
 // TODO: retry the handshake/replay if it fails ?
 func (h *Handshaker) Handshake(ctx context.Context, proxyApp proxy.AppConns) error {
+	return h.HandshakeWithABCIRes(ctx, nil, proxyApp)
+}
+
+func (h *Handshaker) HandshakeWithABCIRes(ctx context.Context, res *abci.InfoResponse, proxyApp proxy.AppConns) error {
 	// Handshake is done via ABCI Info on the query conn.
-	res, err := proxyApp.Query().Info(ctx, proxy.InfoRequest)
-	if err != nil {
-		return fmt.Errorf("error calling Info: %v", err)
+	if res == nil {
+		var err error
+		res, err = proxyApp.Query().Info(ctx, proxy.InfoRequest)
```

Comment:
We need `proxyApp` below when calling `ReplayBlocks`. But I agree that perhaps there is no need to make the call again. We could put the extra call in `Handshake(` if we decide to keep it, in case people are using that function and then pass the retrieved arguments?

Reply:
Ah I didn't see that `proxyApp` is needed below. I would just keep one `Handshake` function, without retrying the call to keep the code as simple as possible. (Unless we need it for example for testing.)

Reply:
I agree with @hvanz in that we probably don't want to retry the call. Logic is clearer like that IMO